script.js

```diff
@@ -0,0 +1,93 @@
+/* ---------- Slideshow ---------- */
+let current_slide = 0;
+let slideNodes = [];
+
+function showSlide(n) {
+  if (!slideNodes.length) return;
+  // wrap-around
+  if (n >= slideNodes.length) current_slide = 0;
+  else if (n < 0) current_slide = slideNodes.length - 1;
+  else current_slide = n;
+
+  for (let i = 0; i < slideNodes.length; i++) {
+    slideNodes[i].style.display = (i === current_slide) ? "block" : "none";
+  }
+}
+
+function previous() { showSlide(current_slide - 1); }
```

Comment:
The previous should not only show the previous. It should change the current_slide value.
`current_slide = current_slide - 1`
`showSlide(current_slide)`